src/main/java/org/spongepowered/api/world/gen/populator/Dungeon.java

```diff
@@ -75,12 +78,50 @@ default void setAttemptsPerChunk(int attempts) {
     }
 
     /**
-     * Gets the {@link MobSpawnerData} which represents the MobSpawner which
-     * will be created within the dungeon.
-     * 
-     * @return The mob spawner data
+     * <p>Gets the {@link MobSpawnerData} which represents the MobSpawner which
+     * will be created within the dungeon.</p>
+     *
+     * <p><b>Note: </b> Only one of choices or mob spawner data
+     * will be present.</p>
+     *
+     * @return The mob spawner data, if present
+     */
+    Optional<MobSpawnerData> getMobSpawnerData();
```

Comment:
As explained in the docs, no. The vanilla system chooses a random String (in the case of the API, `EntityArchetype` to be more flexible) to create the data from. This is different to the MobSpawners "possible entities" as it chooses just one from the list to create. If we were to always force a MobSpawnerData to be present the API would either be inconsistent or lacking. 

So I have made two options - a weighted table of types (which defaults to the unmodifiable vanilla list) with `choices()`, or setting a static MobSpawnerData.